File: plugin/src/org/jetbrains/yesod/lucius/LuciusFileType.java
package org.jetbrains.yesod.lucius;

/**
 * @author Leyla H
 */
import com.intellij.openapi.fileTypes.LanguageFileType;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.haskell.icons.HaskellIcons;
import org.jetbrains.yesod.lucius.LuciusLanguage;

import javax.swing.*;

public class LuciusFileType extends LanguageFileType {
    public static final LuciusFileType INSTANCE = new LuciusFileType();
    @NonNls
    public static final String DEFAULT_EXTENSION = "lucius";

    private Icon myIcon;

    private LuciusFileType() {
        super(LuciusLanguage.INSTANCE);
        myIcon = HaskellIcons.HAMLET;
    }

    @NotNull
    public String getName() {
        return "Lucius file";
    }

    @NotNull
    public String getDescription() {
        return "Lucius file";
    }

    @NotNull
    public String getDefaultExtension() {
        return DEFAULT_EXTENSION;
    }

    public Icon getIcon() {
        return myIcon;
    }

    public String getCharset(@NotNull VirtualFile file, final byte[] content) {
        return "UTF-8";
    }
}

File: plugin/src/org/jetbrains/yesod/lucius/LuciusFileType.kt
package org.jetbrains.yesod.lucius

/**
 * @author Leyla H
 */
import com.intellij.openapi.fileTypes.LanguageFileType
import com.intellij.openapi.vfs.VirtualFile
import org.jetbrains.annotations.NonNls
import org.jetbrains.haskell.icons.HaskellIcons
import org.jetbrains.yesod.lucius.LuciusLanguage

import javax.swing.*

class LuciusFileType private constructor() : LanguageFileType(LuciusLanguage.INSTANCE) {

    private val myIcon: Icon

    init {
        myIcon = HaskellIcons.HAMLET
    }

    override fun getName(): String {
        return "Lucius file"
    }

    override fun getDescription(): String {
        return "Lucius file"
    }

    override fun getDefaultExtension(): String {
        return DEFAULT_EXTENSION
    }

    override fun getIcon(): Icon? {
        return myIcon
    }

    override fun getCharset(file: VirtualFile, content: ByteArray): String? {
        return "UTF-8"
    }

    companion object {
        val INSTANCE: LuciusFileType = LuciusFileType()
        val DEFAULT_EXTENSION: String = "lucius"
    }
}